composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt

```diff
@@ -149,6 +150,20 @@ class Engine(
         proxy = preferences.proxy,
     )
 
+    private fun maxRuntime(
+        taskOrigin: TaskOrigin,
+        netTest: NetTest,
+        preferences: EnginePreferences,
+    ) = if (taskOrigin == TaskOrigin.AutoRun) {
+        MAX_RUNTIME_DISABLED
+    } else if (netTest.callCheckIn == true) {
+        preferences.maxRuntime?.inWholeSeconds?.toInt()?.let { maxRuntimePreference ->
+            if (maxRuntimePreference > 0) 30 + maxRuntimePreference else MAX_RUNTIME_DISABLED
+        } ?: MAX_RUNTIME_DISABLED
+    } else {
+        netTest.test.runtime(inputs = netTest.inputs).inWholeSeconds.toInt()
```

Comment:
Isn't this number for estimation? Ewe shouldn't be using it for max runtime should we? I though it was unlimited unless for WebConnectivity...